controller_manager:
  ros__parameters:
    update_rate: 30
    use_sim_time: true # Uncomment if using simulation

diff_drive:
  ros__parameters:
    type: diff_drive_controller/DiffDriveController

    publish_rate: 50.0

    base_frame_id: base_link

    left_wheel_names: ['left_wheel_joint']
    right_wheel_names: ['right_wheel_joint']
    wheel_separation: 0.18288
    wheel_radius: 0.0325

    use_stamped_vel: false # use stamped velocity commands for Nav2 
    position_feedback: true

    odom_frame_id: odom
    open_loop: false
    enable_odom_tf: true

    publish_odometry: true
    publish_tf: true

    cmd_vel_timeout: 1.0

    # wheels_per_side: x
    # wheel_separation_multiplier: x
    # left_wheel_radius_multiplier: x
    # right_wheel_radius_multiplier: x

    # pose_covariance_diagonal: x
    # twist_covariance_diagonal: x
    
    # publish_limited_velocity: x
    velocity_rolling_window_size: 10
  
    # linear.x.has_velocity_limits: false
    # linear.x.has_acceleration_limits: false
    # linear.x.has_jerk_limits: false
    # linear.x.max_velocity: NAN
    # linear.x.min_velocity: NAN
    # linear.x.max_acceleration: NAN
    # linear.x.min_acceleration: NAN
    # linear.x.max_jerk: NAN
    # linear.x.min_jerk: NAN

    # angular.z.has_velocity_limits: false
    # angular.z.has_acceleration_limits: false
    # angular.z.has_jerk_limits: false
    # angular.z.max_velocity: NAN
    # angular.z.min_velocity: NAN
    # angular.z.max_acceleration: NAN
    # angular.z.min_acceleration: NAN
    # angular.z.max_jerk: NAN
    # angular.z.min_jerk: NAN

joint_broad:
  ros__parameters:
    type: joint_state_broadcaster/JointStateBroadcaster